src/js/utils/parser.js

```diff
@@ -139,18 +139,21 @@ define([
 
     /**
      * Determine the adaptive type
+     * Duration can be positive or negative, but minDvrWindow should always be positive
      */
-    parser.adaptiveType = function(duration) {
+    parser.streamType = function(duration, _minDvrWindow) {
+        var minDvrWindow = _.isUndefined(_minDvrWindow) ? 120 : _minDvrWindow;
+        var streamType = 'VOD';
+
```

Comment:
We could return 'VOD' early if `duration >= 0`. The the only work is to check for 'DVR' conditions otherwise return 'Live'.